### PR TITLE
bpo-40810: Require SQLite 3.7.15

### PR DESCRIPTION
**Pull request by erlend-aasland**
Remove code required to support SQLite pre 3.7.15.

https://bugs.python.org/issue40810

### CODE REVIEW

**Comment by erlend-aasland** [comment]
@berkerpeksag Would you mind reviewing this?

**Review by serhiy-storchaka — APPROVED** [review]
Well, SQLite 3.7.15 was released 8 years ago, so I think we are safe to bump the minimal required version. But please wait for confirmation of other core developer.

**Comment by erlend-aasland** [comment]
Thanks, @serhiy-storchaka! If you've got a suggestion for another core dev that'd be interested in reviewing this, please feel free to request their review.

FYI, rebased onto master for the #24110 NEWS item fix.

**Comment by erlend-aasland** [comment]
@pablogsal Would you mind reviewing this? (Ref. your review of #20909)

**Review by berkerpeksag — APPROVED** [review]
LGTM, but I'd prefer stick to 3.7.15 for a while. I feel like we are getting a bit carried away with dropping support for older SQLite versions :)

**Comment by erlend-aasland** [comment]
> LGTM, but I'd prefer stick to 3.7.15 for a while. I feel like we are getting a bit carried away with dropping support for older SQLite versions :)

Sure, I agree :) There's not too much support code laying around now, so I think we can stay at 3.7.15 for now. Thanks.

**Pull request merged**
